### ISSUE TITLE
Upgrade chart to Support Helm 3

### ISSUE THREAD

**Issue by taylorsilva** [concourse/concourse-chart#56]
With the release of [Helm 3] and v2 of the API, we should look at migrating the chart to the new version. ~Looking to get an idea if people are ready to make that change and/or if there's a reason to hold off on migrating the chart yet. #53 already does this migration so taking it in now would resolve this issue. We want to get an idea if this is something users are okay with.~ **The only negative thing about this change is that you can't use the Helm 2 CLI with the chart once it goes to v2 of the API.**

EDIT: We're pretty far into the lifecyle of Helm 3 now and Helm 2 is only has three months of security patches left. We should cut over now. https://helm.sh/blog/helm-v2-deprecation-timeline/

Notes:
- https://helm.sh/blog/helm-3-released/
- Helm v2 deprecation timeline (6 mo bug/1 yr security): https://helm.sh/blog/2019-10-22-helm-2150-released/#helm-2-support-plan
- Migration tooling: https://helm.sh/blog/migrate-from-helm-v2-to-helm-v3/
- Check out PR #53 for what needs to be updated to be fully helm 3 compatible

[Helm 3]:https://helm.sh/blog/helm-3-released/

I'm unsure if any issues currently do exist when trying to use helm 3 with the current version of the chart.

---
## Goal

Using Helm v3 with the chart works without any issues.

## Tasks

Updating the chart first is probably a good first step. Then let the jobs that fail in CI guide the rest of this story/epic.

- [x] Make necessary changes to concourse-chart so it works with helm 3 (see #53 for pointers)
  - [x] Update `README` to ensure all commands shown work with helm 3
- [x] Update k8s-topgun to run with `helm` v3. In-progress: https://github.com/concourse/concourse/tree/helm3-k8s-topgun
  - concourse/concourse#5219
- [ ] k8s-topgun tests for external postgres and Prometheus should be updated to not use the deprecated `helm/charts` repo 
- [x] Update CI to use helm v3 (docker images, task scripts, etc.)
- [x] Update the [enterprise helm pipeline](https://ci.concourse-ci.org/teams/main/pipelines/enterprise-helm-chart)
  - [x] Package the Helm 3 CLI in the `helm-cli` job
  - [ ]  Update install docs: https://docs.pivotal.io/p-concourse/v6/installation/install-concourse-helm/
- [ ] Migrate our clusters to helm3 using the plugin [2to3](https://github.com/helm/helm-2to3)
  - [ ] hush-house
  - [ ] cluster-1
  - [ ] greenpeace
  - [ ] topgun

**Comment by taylorsilva** [comment]
Working on this.

- Updating our k8s test suite here: https://github.com/concourse/concourse/tree/helm3-k8s-topgun
- Updating ci here: https://github.com/concourse/ci/tree/helm3

**Comment by xtreme-vikram-yadav** [comment]
We ran the tests as is with latest helm(3.3.0) and saw a few tests failing with the error below:

Thrown by *concourse/topgun/k8s/web_scaling_test.go:19*

`
Error: UPGRADE FAILED: failed to replace object: Service "topgun-swi-19146647-postgresql" is invalid: spec.clusterIP: Invalid value: "": field is immutable && failed to replace object: Service "topgun-swi-19146647-web" is invalid: spec.clusterIP: Invalid value: "": field is immutable && failed to replace object: Service "topgun-swi-19146647-web-worker-gateway" is invalid: spec.clusterIP: Invalid value: "": field is immutable
`

This error was thrown by running: 
`helm upgrade --install --force --namespace topgun-swi-19146647 --set=concourse.web.kubernetes.keepNamespaces=false --set=concourse.worker.bindIp=0.0.0.0 --set=postgresql.persistence.enabled=false --set=web.resources.requests.cpu=500m --set=worker.readinessProbe.httpGet.path=/ --set=worker.readinessProbe.httpGet.port=worker-hc --set=worker.resources.requests.cpu=500m --set=image=concourse/concourse-dev --set=imageDigest=sha256:2bdf591fbf87bde9d867a02dfc540ac2eb96ed54d135971e9649b1f1b18e1daa --set=web.replicas=0 --set=worker.replicas=1 topgun-swi-19146647 /Users/pivotal/workspace/concourse-chart/`

After some research, we tried running topgun tests by downgrading to helm 3.1.3 as suggested by https://github.com/helm/helm/issues/7956#issuecomment-619395643. This passed the tests but this could only be used as a temporary solution.

https://github.com/helm/helm/issues/6378#issuecomment-557746499 might provide a better explanation about the error and may lead to a better solution but that investigation to to be continued.
